test/e2e/suite/conformance/certificates/tests.go

```diff
@@ -888,11 +890,21 @@ func (s *Suite) Define() {
 			Expect(err).NotTo(HaveOccurred())
 
 			By("Updating the Certificate after having added an additional dnsName")
-			testCertificate = testCertificate.DeepCopy() // DeepCopy before updating
 			newDNSName := e2eutil.RandomSubdomain(s.DomainSuffix)
-			testCertificate.Spec.DNSNames = append(testCertificate.Spec.DNSNames, newDNSName)
+			retry.RetryOnConflict(retry.DefaultRetry, func() error {
```

Comment:
Suggestion: instead of retrying 'Update' operation, perhaps we could use 'Patch' - that way there we could avoid conflicts without needing to retry multiple times. I think this would make sense as we are moving to server side apply/patching in other places too, so this would be a similar solution.
I've done that in integration tests for exponential backoff PR i.e [here](https://github.com/irbekrm/cert-manager/blob/exp_backoff/test/integration/certificates/trigger_controller_test.go#L414)

Reply:
It took me 20 minutes just to remember how the server-side apply works 😇😬 I still do not fully grasp it, for example I don't know whether the whole object gets diff'd with the server's object and then the fields that don't match become owned by the `test` controller, but I'll go with it

Reply:
https://github.com/cert-manager/cert-manager/issues/4948